src/audio/module_adapter/module/generic.c

```diff
@@ -197,15 +197,13 @@ int module_prepare(struct processing_module *mod)
 
 	comp_dbg(dev, "module_prepare() start");
 
-	if (mod->priv.state == MODULE_IDLE)
-		return 0;
 	if (mod->priv.state < MODULE_INITIALIZED)
 		return -EPERM;
 
 	ret = md->ops->prepare(mod);
 	if (ret) {
-		comp_err(dev, "module_prepare() error %d: module specific prepare failed, comp_id %d",
-			 ret, dev_comp_id(dev));
+		comp_err(dev, "module_prepare() error %d: module specific prepare failed, comp_id %din state %d",
+			 ret, dev_comp_id(dev), md->state);
```

Comment:
This change should go in a different patch as it is not strictly related to current patch.

Reply:
Before this patch, module specific prepare() is called only in MODULE_INITIALIZED state.
After this patch, it will be called in MODULE_INITIALIZED and MODULE_IDLE states. So it becomes crucial to show the state in error message.

Reply:
@johnylin76 yes, but this can be done in a separate patch.